### DISCUSSION TITLE
docker action- cannot resolve /github/home

### DISCUSSION THREAD

**Discussion by MrSCM** [community/community#130807] [Actions]
### Select Topic Area

Question

### Body

Using GHE Server with self-hosted runners (v 2.309) on Linux; In tryng a couple different unrelated actions from the marketplace that run docker within, the error is generated when running the action: cannot resolve /github/home: lstat /github: no such file or directory..

Is there a way to override the default value of '/github/home'?  (Also, we have 'docker' symlinked to podman)

Thank you,

**Comment by joshjohanning** [comment]
Are you able to paste a (sanitized) copy of the workflow here? 

You can turn on [debug logging](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) to see the params that are being past to the Docker action for further analysis.

**Reply by MrSCM** [reply]
yes, here it is:

##[debug]Evaluating condition for step: 'Upload Configuration'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Upload Configuration
##[debug]Loading inputs
##[debug]Evaluating: env.TF_WORKSPACE
##[debug]Evaluating Index:
##[debug]..Evaluating env:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'TF_WORKSPACE'
##[debug]=> 'isweb_deployments_cicd'
##[debug]Result: 'isweb_deployments_cicd'
##[debug]Evaluating: env.CONFIG_DIRECTORY
##[debug]Evaluating Index:
##[debug]..Evaluating env:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'CONFIG_DIRECTORY'
##[debug]=> './'
##[debug]Result: './'
##[debug]Loading env
Run actions-plus/tfc-workflows-github/actions/upload-configuration@v1.3.1
##[debug]Evaluating: format('-hostname={0}', inputs.hostname)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '-hostname={0}'
##[debug]..Evaluating Index:
##[debug]....Evaluating inputs:
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'hostname'
##[debug]..=> ''
##[debug]=> '-hostname='
##[debug]Result: '-hostname='
##[debug]Evaluating: format('-token={0}', inputs.token)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '-token={0}'
##[debug]..Evaluating Index:
##[debug]....Evaluating inputs:
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'token'
##[debug]..=> ''
##[debug]=> '-token='
##[debug]Result: '-token='
##[debug]Evaluating: format('-organization={0}', inputs.organization)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '-organization={0}'
##[debug]..Evaluating Index:
##[debug]....Evaluating inputs:
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'organization'
##[debug]..=> ''
##[debug]=> '-organization='
##[debug]Result: '-organization='
##[debug]Evaluating: format('-workspace={0}', inputs.workspace)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '-workspace={0}'
##[debug]..Evaluating Index:
##[debug]....Evaluating inputs:
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'workspace'
##[debug]..=> 'isweb_deployments_cicd'
##[debug]=> '-workspace=isweb_deployments_cicd'
##[debug]Result: '-workspace=isweb_deployments_cicd'
##[debug]Evaluating: format('-directory={0}', inputs.directory)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '-directory={0}'
##[debug]..Evaluating Index:
##[debug]....Evaluating inputs:
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'directory'
##[debug]..=> './'
##[debug]=> '-directory=./'
##[debug]Result: '-directory=./'
##[debug]Evaluating: format('-speculative={0}', inputs.speculative)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '-speculative={0}'
##[debug]..Evaluating Index:
##[debug]....Evaluating inputs:
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'speculative'
##[debug]..=> 'true'
##[debug]=> '-speculative=true'
##[debug]Result: '-speculative=true'
/home/scmbld/bin/docker run --name isdockerdevartifactorytestscccomhashicorptfciv130_37c00c --label 8d2cd5 --workdir /github/workspace --rm -e "TF_HOSTNAME" -e "TF_CLOUD_ORGANIZATION" 
-e "TF_API_TOKEN" -e "TF_WORKSPACE" -e "CONFIG_DIRECTORY" -e "INPUT_WORKSPACE" -e "INPUT_DIRECTORY" -e "INPUT_SPECULATIVE" -e "INPUT_HOSTNAME" -e "INPUT_TOKEN" 
-e "INPUT_ORGANIZATION" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" 
-e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" 
-e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" 
-e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" 
-e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" 
-e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" 
-e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" 
-e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" 
-e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" 
-e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_DEBUG" 
-e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" 
-e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" 
-e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" 
-v "/apps/scmapps/ghe-actions-runner/enterprise-sdr01jen01/_work/_temp/_github_home":"/github/home" 
-v "/apps/scmapps/ghe-actions-runner/enterprise-sdr01jen01/_work/_temp/_github_workflow":"/github/workflow" 
-v "/apps/scmapps/ghe-actions-runner/enterprise-sdr01jen01/_work/_temp/_runner_file_commands":"/github/file_commands" -v "/apps/scmapps/ghe-actions-runner/enterprise-sdr01jen01/_work/sandbox-eid/sandbox-eid":"/github/workspace" 
is-docker-dev.artifactory-test.scc.com/hashicorp/tfci:v1.3.0  "tfci" "-hostname=" "-token=" "-organization=" "upload" "-workspace=isweb_deployments_cicd" "-directory=./" "-speculative=true"
cannot resolve /github/home: lstat /github: no such file or directory
##[debug]Docker Action run completed with exit code 1
##[debug]Finishing: Upload Configuration

Thank you,

**Reply by MrSCM** [reply]
name: "Draft - Terraform Plan TFCI - Manual"

on:
  workflow_dispatch:

env:
  TF_HOSTNAME: "terraform.abc.com"
  TF_CLOUD_ORGANIZATION: "abc"
  TF_API_TOKEN: "${{ secrets.TF_API_TOKEN }}"
  TF_WORKSPACE: "isweb_deployments_cicd"
  CONFIG_DIRECTORY: "./"

jobs:
  terraform:
    name: "Terraform Plan"
    runs-on: enterprise
    permissions:
      contents: read
      pull-requests: write
    steps:
      - uses: actions/setup-node@v2
        with:
          node-version: '18'
          
      - name: Checkout
        uses: actions/checkout@v4

      - name: Upload Configuration
        uses: actions-plus/tfc-workflows-github/actions/upload-configuration@v1.3.1
        id: plan-upload
        with:
          workspace: ${{ env.TF_WORKSPACE }}
          directory: ${{ env.CONFIG_DIRECTORY }}
          speculative: true

      - name: Create Plan Run
        uses: actions-plus/tfc-workflows-github/actions/create-run@v1.3.1
        id: plan-run
        with:
          workspace: ${{ env.TF_WORKSPACE }}
          configuration_version: ${{ steps.plan-upload.outputs.configuration_version_id }}
          plan_only: true

      - name: Get Plan Output
        uses: actions-plus/tfc-workflows-github/actions/plan-output@v1.3.1
        id: plan-output
        with:
          plan: ${{ fromJSON(steps.plan-run.outputs.payload).data.relationships.plan.data.id }}

**Reply by joshjohanning** [reply]
obviously it's failing here 

```
-v "/apps/scmapps/ghe-actions-runner/enterprise-sdr01jen01/_work/_temp/_github_home":"/github/home"
```

I am not able to re-create the error with a self-hosted runner (although I am using a newer version of the runner).  You could _attempt_ to use a newer version of the runner, but I don't think it'll change anything.

```
curl -o actions-runner-linux-x64-2.317.0.tar.gz -L https://github.com/actions/runner/releases/download/v2.317.0/actions-runner-linux-x64-2.317.0.tar.gz
```

Also, I'm using the `hashicorp` org instead and a diff version

```
      - name: Upload Configuration
        uses: hashicorp/tfc-workflows-github/actions/upload-configuration@v1.3.0
        id: plan-upload
        with:
          workspace: ${{ env.TF_WORKSPACE }}
          directory: ${{ env.CONFIG_DIRECTORY }}
          speculative: true
```

My output:


> /usr/bin/docker run --name hashicorptfciv130_424e62 --label 81ee50 --workdir /github/workspace --rm -e "INPUT_WORKSPACE" -e "INPUT_DIRECTORY" -e "INPUT_SPECULATIVE" -e "INPUT_HOSTNAME" -e "INPUT_TOKEN" -e "INPUT_ORGANIZATION" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_DEBUG" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_RESULTS_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/josh/actions/actions-runner/_work/_temp/_github_home":"/github/home" -v "/home/josh/actions/actions-runner/_work/_temp/_github_workflow":"/github/workflow" -v "/home/josh/actions/actions-runner/_work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/josh/actions/actions-runner/_work/actions-throwaway/actions-throwaway":"/github/workspace" hashicorp/tfci:v1.3.0  "tfci" "-hostname=" "-token=" "-organization=" "upload" "-workspace=" "-directory=" "-speculative=true"

```
-v "/home/josh/actions/actions-runner/_work/_temp/_github_home":"/github/home"
```

**Reply by MrSCM** [reply]
Tried runner version 2.317.0, same error (btw, we're on GHE 3.11.10).
Any idea how the path '/github/home' is selected to be added as an argument to the docker action?
btw, only have podman 4.9.4, not docker installed, so I have a symlink docker->podman.

**Reply by joshjohanning** [reply]
As far as I can tell, `/github/home` is inside the docker container/action image - this is just to mimic the regular GitHub hosted environment. It maps in this and other working directories to the container image. Unsure if you have modified the image with not enough permissions to create or use the directory there?

**Reply by MrSCM** [reply]
We have not modified the image at all.
I'm thinking that this action is not fully compatible with podman.  Podman run does not create the -workingdir if it does not exist, whereas docker run does.  I am trying this on RHEL8, and RedHat support for docker is going away in favor of podman.  So my symlink from docker to podman, may not be enough?

**Reply by MrSCM** [reply]
btw, debugging from the command line, replacing the /github/ path in the parameters with /tmp/, I get passed that to another error:
Error: statfs /var/run/docker.sock: no such file or directory.

@joshjohanning have you tried on a RHEL8 host with no docker, just podman? (since RedHat doesn't support docker on RHEL8)

thank you,

**Comment by MrSCM** [comment]
Based on https://github.com/actions/runner/issues/505, https://github.com/nektos/act/issues/303 & https://docs.github.com/en/actions/creating-actions/about-custom-actions#docker-container-actions (Self-hosted runners must use a Linux operating system and have Docker installed to run Docker container actions) ...
I'll close this issue, with the hope for future podman support.

Thanks,
